python/hail/methods/misc.py

```diff
@@ -244,7 +244,7 @@ def rename_duplicates(dataset, name='unique_id') -> MatrixTable:
     return MatrixTable(dataset._jvds.renameDuplicates(name))
 
 
-@typecheck(ds=MatrixTable,
+@typecheck(ds=oneof(Table, MatrixTable),
            intervals=expr_array(expr_interval(expr_any)),
            keep=bool)
 def filter_intervals(ds, intervals, keep=True) -> MatrixTable:
```

Comment:
Will this fail the typechecker if it returns a Table?

Reply:
No, Python type hints aren't enforced, but I changed it to Union[Table, MatrixTable].